codegen.cpp

```diff
@@ -22,16 +22,28 @@ void NodeAdd::codegen() {
     lhs->codegen();
     cout << "  str x0, [sp, -16]!" << endl;
     rhs->codegen();
+    if(c_type->isPtrType()) {
```

Comment:
This check is wrong, should be checking lhs->c_type and rhs->c_type to implement all 4 cases.